application/client/src/app/ui/views/toolbar/chart/zoomer/cursor/component.ts

```diff
@@ -0,0 +1,50 @@
+import {
+    Component,
+    HostListener,
+    AfterViewInit,
+    ChangeDetectorRef,
+    ViewContainerRef,
+    Input,
+} from '@angular/core';
+import { Ilc, IlcInterface } from '@env/decorators/component';
+import { ChangesDetector } from '@ui/env/extentions/changes';
+import { State } from './state';
+
+@Component({
+    selector: 'app-views-chart-zoomer-cursor-canvas',
+    templateUrl: './template.html',
+    styleUrls: ['./styles.less'],
+})
+@Ilc()
+export class ViewChartZoomerCursorCanvas extends ChangesDetector implements AfterViewInit {
+    @Input() getLeftOffset: () => number = () => 0;
+
+    public state: State = new State();
+
+    constructor(cdRef: ChangeDetectorRef, private _vcRef: ViewContainerRef) {
+        super(cdRef);
+        window.addEventListener('mousemove', this.state.onWindowMousemove);
+        window.addEventListener('mouseup', this.state.onWindowMouseup);
```

Comment:
you are linking window events to state methods.In destroy handler bellow you are unsubscribing from that. Would it be better just move it into state? You have already `state.init(...)` well subscription could happen there